server/graphql/root-query.js

```diff
@@ -0,0 +1,107 @@
+import {
+	GraphQLString,
+	GraphQLObjectType,
+	GraphQLNonNull
+} from 'graphql';
+import jwt from 'jsonwebtoken';
+import bcrypt from 'bcrypt-nodejs';
+import uuid from 'node-uuid';
+import UserType from './user';
+import mongoose from 'mongoose';
+
+import { User as UserModel } from '../models';
+
+const secretKey = uuid.v4();
+
+function authenticateUser(username, password, user) {
+	if (!user) {
+		return false;
+	}
+
+	try {
+		return bcrypt.compareSync(password, user.password);
+	} catch (e) {
+		console.error(e);
+		return false;
+	}
+}
+
+export default new GraphQLObjectType({
+	name: 'RootQueries',
+	fields: () => ({
+		echo: {
+			type: GraphQLString,
```

Comment:
Every indentation is only 1 tab character, but github interprets them as 8 spaces